tests/epyccel/test_epyccel_sign.py

```diff
@@ -0,0 +1,185 @@
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+
+def test_sign_int32(language):
+    f_pos = mod.int32_pos
+    f_neg = mod.int32_neg
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+
+def test_sign_int64(language):
+    f_pos = mod.int64_pos
+    f_neg = mod.int64_neg
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+
+def test_sign_float(language):
+    f_pos = mod.float_pos
+    f_neg = mod.float_neg
+    f_nul = mod.float_nul
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+    f_nul_epyc = epyccel(f_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+
+def test_sign_float64(language):
+    f_pos = mod.float64_pos
+    f_neg = mod.float64_neg
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+
+def test_sign_literal_complex(language):
+    f_pos      = mod.literal_complex_pos
+    f_neg      = mod.literal_complex_neg
+    f_nul      = mod.literal_complex_nul_nul
+    f_nul_imag = mod.literal_complex_nul_imag
+    f_real_nul = mod.literal_complex_real_nul
+    f_pos_epyc      = epyccel(f_pos, language = language)
+    f_neg_epyc      = epyccel(f_neg, language = language)
+    f_nul_epyc      = epyccel(f_nul, language = language)
+    f_nul_imag_epyc = epyccel(f_nul_imag, language = language)
+    f_real_nul_epyc = epyccel(f_real_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+    assert f_nul_imag_epyc() == f_nul_imag()
+    assert f_real_nul_epyc() == f_real_nul()
+
+def test_sign_literal_int(language):
+    f_pos = mod.literal_int_pos
+    f_neg = mod.literal_int_neg
+    f_nul = mod.literal_int_nul
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+    f_nul_epyc = epyccel(f_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+
+def test_sign_literal_float(language):
+    f_pos = mod.literal_float_pos
+    f_neg = mod.literal_float_neg
+    f_nul = mod.literal_float_nul
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+    f_nul_epyc = epyccel(f_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
```

Comment:
These tests are also in this file and don't check dtype

Reply:
I tried the following test and I was getting a weird behavior, if you can explain why that happens.
```python
import numpy as np

if __name__ == "__main__":
    b = np.sign(np.complex64(1+4j))
    print(b)
```

Inside the function `_print_NumpySign` I was printed at the beginning the `expr.args` and, logically speaking, I was expecting an object of type `NumpyComplex64` but I got `(Literal((1+4j)),)`

Reply:
`NumpyComplex64` is a cast function. Casting a literal to a specific precision is equivalent to keeping the literal, but changing the precision

Reply:
Well the kind of the real part of the complex number says `f64` instead of `f32`

Reply:
> Well the kind of the real part of the complex number says `f64` instead of `f32`. `(1.0_f64, 5.0_f64)`

That is indeed a bug. `numpy.complex64` is a complex number type composed of two 32-bit-precision floating-point numbers

Reply:
Although this is a bug, it shouldn't lead to wrong results so you don't need to worry about it for this PR

Reply:
Tests were failing when I was checking for dtype, specifically for `numpy.complex64/128`

Reply:
I suppose I should use that `get_final_precision` function too in the `NumpySign`.

Reply:
I suspect so, but you need to check the behaviour of `NumpySign`. `get_final_precision` keeps the precision of the object you pass to it, but converts it to a positive value. This means, for example, that the function can't return a `float` (precision=-1), only a `float32` or `float64`. `default_precision` contains the numpy precision of the pure python type (`int`, `float`, `complex`)